Add cmd management to http.py

AnotherCat opened the pull request:
## About this pull request
This is needed because previously `utils.manage_commands` functions were used in `auto_register` and `auto_delete` and those functions created an aiohttp session for each function, which to quote aiohttp's docs "making a session for every request is a **very bad** idea."
This creates functions in `http.SlashCommandRequest` that use discord's session to make requests.

## Changes
Adds `add_slash_command`, `remove_slash_command`, `get_all_commands` and `command_request` to `SlashCommandRequest`
Changes `register_all_commands` and `delete_unused_commands` to use the new functions.
- [x] I checked this pull request runs on `Python 3.6.X`.
- [ ] This fixes something in [Issues](https://github.com/eunwoo1104/discord-py-slash-command/issues).
    - Issue:
- [ ] This adds something new.
- [ ] There is/are breaking change(s).
- [x] (If required) Document string is updated/added.
- [ ] This changes anything except python code. (README, docs, etc.)

eunwoo1104 commented:
Leaving this as a external comment since it requires a lot of changes.
We already have discord client object. So we just can get bot's id as `self._discord.user.id` and remove all `bot_id` param.

AnotherCat commented:
I did notice that, but i thought that because all the other functions in `SlashCommandRequest` took `bot_id` as a parameter i assumed that i should for these functions.
I've:
- Remove `bot_id` parameter from all functions in `SlashCommandRequest`
- Updated all places where http functions were called
- Removed `bot_id` from `SlashMessage` as it was only used for http functions

eunwoo1104 commented:
> I did notice that, but i thought that because all the other functions in `SlashCommandRequest` took `bot_id` as a parameter i assumed that i should for these functions.

Oh, I thought I fixed that, but seems like I completely forgotten about it. Apology for that.
Anyways looks fine now. Merging this pull request, thank you for contributing!
